source/components/molecules/ImageUploader/ImageUploader.tsx

```diff
@@ -124,18 +124,23 @@ const ImageUploader: React.FC<Props> = ({
 }) => {
   const [choiceModalVisible, toggleModal] = useModal();
 
-  const renameImagesWithPrefix = (images: Image[], prefix: string) =>
-    images.map((image, idx) => ({
-      ...image,
-      filename: `${prefix}_${idx}`,
-    }));
+  const renameImageWithAffix = (
+    image: Image,
+    prefix: string,
+    suffix: string
+  ) => ({
+    ...image,
+    filename: `${prefix}_${suffix}`,
+  });
```

Comment:
The usage of `prefix` and `suffix` is not entirely correct. `suffix` and `prefix` is used to add characters before or after a base word. `prefix`-`base word`- `suffix`. So in in your case the more correct logic would be:

```js
  const renameImageWithSuffix = (
    image: Image,
    imageName: string,
    suffix: string
  ) => ({
    ...image,
    filename: `${imageName}_${suffix}`,
  });


if (...) {
  updatedImages = updatedImages.map((image, index) =>
    renameImageWithSuffix(image, imageName, index.toString())
   );
}
```